### PR TITLE
[Server] Add support for deleting multiple patterns without exploiting requests

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
**Description**
When deleting multiple patterns at once

Instead of sending 100 requests to server for deleting 100 patterns we can try sending one request with 100ids to the database.

This PR fixes https://github.com/meshery/meshery/issues/4158

https://user-images.githubusercontent.com/48255244/135865108-224a71d8-6c33-4a64-a665-04f523bf5556.mp4

**Notes for Reviewers**
Meshery-cloud: https://github.com/layer5io/meshery-cloud/pull/128

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Uh-oh, a merge conflict to look into...

File: models/meshery_pattern_persister.go [hunk 1]
```diff
@@ -62,6 +63,17 @@ func (mpp *MesheryPatternPersister) DeleteMesheryPattern(id uuid.UUID) ([]byte,
...
+func (mpp *MesheryPatternPersister) DeleteMesheryPatterns(/*ids []uuid.UUID*/) ([]byte, error) {
+	// for _, id := range(ids) {
+	// 	pattern := MesheryPattern{ID: &id}
+	// 	mpp.DB.Delete(&pattern)
+	// }
+	logrus.Info("Came here did nothing-.................")
+
+	return nil, nil // ...
+}
+
```
Comment (Abhishek-kumar09) [inline models/meshery_pattern_persister.go:76]: What is pattern-persistor, I couldn't reach to this code-block while making normal requests, but was throwing error if there is no such function signature. Is is cache or something related? How could I reach the code-block?

File: models/remote_provider.go [hunk 2]
```diff
@@ -1117,6 +1117,53 @@ func (l *RemoteProvider) DeleteMesheryPattern(req *http.Request, patternID strin
...
+	ep, _ := l.Capabilities.GetEndpointForFeature(PersistMesheryPatterns)
+
+	// convert to buffer
+	bf := bytes.NewBufferString(patterns)
+
+	// Create remote provider-url
+	remoteProviderURL, _ := url.Parse(fmt.Sprintf("%s%s", l.RemoteProviderURL, ep))
+	logrus.Debugf("constructed pattern url: %s", remoteProviderURL.String())
+
+	cReq, _ := http.NewRequest(http.MethodDelete, "http://localhost:9876/patterns", bf)
```
Comment (utkarsh-pro) [inline models/remote_provider.go:1136]: Change the URL here

File: models/meshery_pattern_persister.go (hunk 1 above)
Comment (utkarsh-pro) [reply models/meshery_pattern_persister.go]: This is where it is defined https://github.com/meshery/meshery/pull/4324/files/84e0fed1f908a03c41e4daf11fbaf005ab0ec40d#diff-b008d466a53041b957f27ef5d4cf0a6cb5a7b35860fe6511555d67ea3154cc21R14

**Comment by leecalcote** [comment]
@utkarsh-pro ready for review

**Comment by utkarsh-pro** [comment]
@leecalcote I think we need to bring in https://github.com/layer5io/meshery-cloud/pull/128 first before we merge this PR.

**Review by utkarsh-pro — APPROVED** [review]
LGTM

**Comment by sudo-NithishKarthik** [comment]
@Abhishek-kumar09 , merge conflicts to be addressed ☝️

**Comment by leecalcote** [comment]
Good to go here after merge conflict is addressed. ✅

**Comment by Abhishek-kumar09** [comment]
@leecalcote 
Conflicts resolved, good to go after merging changes in meshery-cloud https://github.com/layer5io/meshery-cloud/pull/128

**Review by warunicorn19 — APPROVED** [review]
merge at-will @Abhishek-kumar09

**Pull request merged**